src/App.tsx

```diff
@@ -1,25 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const total = items.length;
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [itemsPerPage, setItemsPerPage] = useState<number>(5);
+
+  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setCurrentPage(1);
+    setItemsPerPage(+e.target.value);
+  };
+
+  const startIndex = (currentPage - 1) * itemsPerPage;
+  const lastIndex = startIndex + itemsPerPage;
+  const visibleItems = items.slice(startIndex, lastIndex);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {startIndex + 1} -{' '}
+        {lastIndex > total ? total : lastIndex} of {total})
```

Comment:
The current logic to display the range of items might show incorrect values if `lastIndex` exceeds `total`. To prevent this, you can calculate the end index for display as `Math.min(lastIndex, total)` outside the JSX and then use it here.